### ISSUE TITLE
nodeport does not work with OVS/OVN 2.9

### ISSUE THREAD

**Issue by shettyg** [openvswitch/ovn-kubernetes#272]
The fix is posted for review here: 
https://mail.openvswitch.org/pipermail/ovs-dev/2018-March/345487.html

The fix should come with OVS 2.9.1

CC: @alinbalutoiu

**Comment by alinbalutoiu** [comment]
Thanks @shettyg !
cc: @aserdean

**Comment by mihail-velikov** [comment]
Hello,

I have also been experiencing issues when trying to enable nodeport on the minion nodes using OVS2.9. For example when trying to ping from a pod to the external world only every second packet is received.  After removing -nodeport from the ovnkube it started working as expected. 

Correct me if I am wrong but without the -nodeport setting it is not possible to expose and internal service to the outside world?

Kind Regards,
Mihail Velikov

**Comment by shettyg** [comment]
@mihail-velikov 

Your observations could be because of a different bug. Let me try and narrow the issue down. So, without -nodeport set, can your pods reach external world when they initiate the connection?

**Comment by mihail-velikov** [comment]
Yes, when i skip the "-nodeport" parameter for the ovnkube command on the worker node the network works as expected.

When I add the parameter - the network still works internally ( i can ping the other hosts internal IPs and such) but for example when I try to ping external IP I get around 70% packet loss. At first the behavior is to get reply for every second packet (example , sequence number 2-4-6-8). After around 30 second I lose my ssh session. While I am still able to ping the external IP the ssh session times out.

**Comment by shettyg** [comment]
@mihail-velikov

Are you running the python ovn-k8s-gateway-helper? Till #258 gets in, we have a requirement that we run the python ovn-k8s-gateway-helper for -nodeport to work.

e.g:
https://github.com/openvswitch/ovn-kubernetes/blob/master/vagrant/provisioning/setup-k8s-minion.sh#L96

**Comment by mihail-velikov** [comment]
No, I have only tried to use the golang implementation as described in the READ.me file. I want to understand if the problem I am experiencing is the same as the one for which this issue was opened. 

The python code is working as expected in our previous setup.

**Comment by shettyg** [comment]
>No, I have only tried to use the golang implementation as described in the READ.me file. I want to understand if the problem I am experiencing is the same as the one for which this issue was opened.

The problem you are experiencing is not because of this bug. This bug does make OVS 2.9.0 untenable. You should use 2.8.1 (For e.g: https://github.com/openvswitch/ovn-kubernetes/blob/master/vagrant/provisioning/setup-minion.sh#L67) 

And for your particular problem, you will have to use the  python ovn-k8s-gateway-helper in addition to ovnkube till we get #258 in.

**Comment by mihail-velikov** [comment]
Ok, I will try tomorrow with 2.8.1 and the ovn-k8s-gateway-helper. Thanks

**Comment by shettyg** [comment]
#258 got merged. So you likely don't need to use the python code anymore. Just ovnkube will do - the only difference after commit 258 is that ovnkube on nodes will not exit when -nodeport is provided - so you will have to run it in background.

**Comment by mihail-velikov** [comment]
Hello @shettyg,

It seems that the pull request that got merged fixed our problems with the intermittent network connectivity that I mentioned. The ping tests that I was doing are working as expected.

Still I am not able to expose the service to the outside world. I can see:
10.31.0.0 is my datacenter network
10.32.0.0 is my openvswitch network
```
Mar 28 06:24:35 lnx-4 ovn-nbctl: ovs|00001|nbctl|INFO|Called as ovn-nbctl set load_balancer 4d48a81d-2649-48b8-b6ce-9373238f6f61 "vips:\"10.31.0.35:31492\"=\"10.32.2.4:80\""
Mar 28 06:24:35 lnx-4 ovn-nbctl: ovs|00001|nbctl|INFO|Called as ovn-nbctl set load_balancer cb7f2526-7042-4a4e-bc9b-dca9ba3802a5 "vips:\"10.32.200.24:80\"=\"10.32.2.4:80\""
```

From inside the openvswitch network connecting to the internal service works as expected:
```
[06:25:41] root@lnx-5:[~]: telnet 10.32.200.24 80
Trying 10.32.200.24...
Connected to 10.32.200.24.
Escape character is '^]'.
^]
```
From my machine:
```
➜  ~ telnet 10.31.0.35 31492
Trying 10.31.0.35...
telnet: Unable to connect to remote host: Operation timed out
➜  ~ ping 10.31.0.35
PING 10.31.0.35 (10.31.0.35): 56 data bytes
64 bytes from 10.31.0.35: icmp_seq=0 ttl=61 time=1.506 ms
64 bytes from 10.31.0.35: icmp_seq=1 ttl=61 time=1.050 ms
^C
--- 10.31.0.35 ping statistics ---
2 packets transmitted, 2 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 1.050/1.278/1.506/0.228 ms
➜  ~ telnet 10.31.0.35 22
Trying 10.31.0.35...
Connected to 10.31.0.35.
Escape character is '^]'.
SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.4
^C
Connection closed by foreign host.
```
For these test I am still running openvswitch 2.9.0. I will try now with 2.8.1 and I will let you know how it goes.

Thank you.

Kind Regards,
Mihail Velikov

**Comment by mihail-velikov** [comment]
Hello again,

I have tested with openvswitch 2.81 and ovnkube only and everything works properly! Thanks a lot!

I will start working on the windows nodes now :)

Kind Regards,
Mihail Velikov

**Comment by shettyg** [comment]
@mihail-velikov

From what I understand, Windows needs OVS 2.9. Since gateways are not installed on Windows nodes, that should be okay.

**Comment by alinbalutoiu** [comment]
@mihail-velikov for the Windows node you could use the following guide from the official kubernetes docs. The doc will be sent upstream, but at moment you can find them here: https://github.com/alinbalutoiu/kubernetes.github.io/commit/92a12e605199070088b42b3a711b082a0fdc9cc1

**Comment by alinbalutoiu** [comment]
@mihail-velikov what is your WIndows Server version? Run from cmd/powershell: `wmic os get buildnumber`

**Comment by mihail-velikov** [comment]
@alinbalutoiu - here is the build number
BuildNumber
14393

Also please note that on the windows node I am running openvswitch 2.7. Another thing that I found is that when I install the cloudbase openvswitch package it automatically adds a ovs-watcher service. I have since disabled this service and rebooted the machine but it didn't help.

**Comment by alinbalutoiu** [comment]
That would be Windows Server 2016 RTM, we are trying now to update the old way that we used to do the things on that version.
You need this line in kubernetes: https://github.com/alinbalutoiu/kubernetes/commit/a994cb531a9e7cd01091997b1daa34e4f7941881#diff-e6e2181e7a7b69ff9cebf666c4ae1989R142
And this update to the CNI which adds the support for Windows Server 2016 RTM in this repo: https://github.com/alinbalutoiu/ovn-kubernetes/commit/5972e7bb30be0d70644cf44d7289edc13502b982
This would be the way that I'd recommend to go on Windows Server 2016 RTM. If you just want to deploy for testing you can use the old CNI from the upstream docs https://kubernetes.io/docs/getting-started-guides/windows/
Unfortunately on this specific Windows version there is a limitation and you cannot have the custom IP/MAC that OVN will assign to you, that will be managed by docker on Windows. There is also the limitation of one container per pod. In the newer Windows versions they added more flexibility and we can manage the container network endpoints from the CNI, thus controlling the IP/MAC address.

**Comment by mihail-velikov** [comment]
@alinbalutoiu - thanks a lot for the information.

Could you just confirm that on Windows 1709 I can use this installer file for OVS?
https://cloudbase.it/downloads/openvswitch-hyperv-2.7.0-certified.msi

The alternative is to compile openvswitch 2.9 from source and see if it works.

Thank you.

Kind Regards,
Mihail Velikov

**Comment by aserdean** [comment]
@mihail-velikov please use https://cloudbase.it/downloads/openvswitch-hyperv-installer-beta.msi if all possible (it has a few bug fixes).

Note on 1709 you will require Hyper-V feature to install and run it properly (they broke backwards compatibility between RTM and RS3). I am still investigating how to add support for it.

**Comment by mihail-velikov** [comment]
Hello all,

In the past days I have tried to use Windows 1709 following these instructions:
https://github.com/alinbalutoiu/kubernetes.github.io/blob/92a12e605199070088b42b3a711b082a0fdc9cc1/docs/getting-started-guides/windows/index.md

Unfortunately I am not able to run it properly and it fails. As the machine does not have network connectivity I have attached several screenshots from the console:

https://imgur.com/a/lhwHG

Any ideas would be greatly appreciated :)

Kind Regards,
Mihail Velikov

**Comment by mihail-velikov** [comment]
It seems I have been installing Docker incorrectly - I was using the commands for Windows 2016 instead the ones for the insider build. I will try now with the correct ones to see if it makes a difference.

Kind Regards,
Mihail Velikov

**Comment by alinbalutoiu** [comment]
Hi @mihail-velikov , there is a small mistake in the script, sorry about that. Can you change the single quotes to double quote? The variable does not get replaced in powershell if it's between single quotes and you can see that in ``ovs-vsctl show``. It adds "$INTERFACE_ALIAS" under br-ex instead of adding the physical interface.
Replace ``'$INTERFACE_ALIAS'`` with ``"$INTERFACE_ALIAS"`` in the command ``ovs-vsctl --no-wait add-port br-ex '$INTERFACE_ALIAS'``.
I will also update the documentation, thanks!

**Comment by mihail-velikov** [comment]
Hello @alinbalutoiu 

I have tried that without luck:
https://imgur.com/a/LtzdH

It seems that the br-ext adapter does not get created for some reason. Also my IP address on my Ethernet interface is manually configured and not through DHCP - do I need to set this up on the bridge?

THank you.

Kind Regards,
Mihail Velikov

**Comment by aserdean** [comment]
@mihail-velikov weird. Do you have Hyper-V feature installed on 1709?
@mihail-velikov can you also show the command for `ovs-dpctl show` also mind sharing the vswitchd.logs?

**Comment by mihail-velikov** [comment]
@aserdean - agree -it is quite strange :)
ovs-dpctl show:
https://imgur.com/a/sjbZN

hyper-v feature:
https://imgur.com/a/Jxq3e

ovs-vswitchd.log
https://imgur.com/a/gA50X

other info:
https://imgur.com/a/sKrP4

**Comment by mihail-velikov** [comment]
Here is my install procedure:
##disable the firewall
Set-NetFirewallProfile -Profile Domain,Public,Private -Enabled False
##Install Hyper-V
Install-WindowsFeature -Name Hyper-V -IncludeManagementTools

##Remove docker if exists
Stop-Service docker
Uninstall-Package docker
Uninstall-Module DockerMsftProvider

##Install docker for windows preview
Install-Module -Name DockerProvider -Repository PSGallery -Force
Install-Package -Name docker -ProviderName DockerProvider -RequiredVersion Preview
Restart-Computer -Force

##Install openvswitch
mkdir C:\ovs
cd C:\ovs
Start-BitsTransfer  https://cloudbase.it/downloads/openvswitch-hyperv-installer-beta.msi
cmd /c 'msiexec /i openvswitch-hyperv-installer-beta.msi ADDLOCAL="OpenvSwitchCLI,OpenvSwitchDriver,OVNHost" /qn'

##Put this script ins a file and run it
$INTERFACE_ALIAS="Ethernet"
Stop-Service ovs-vswitchd -force; Get-VMSwitch -SwitchType External | Disable-VMSwitchExtension "Cloudbase Open vSwitch Extension"
Get-VMSwitch -SwitchType External | Set-VMSwitch -AllowManagementOS $false
# Ignore the error from the first command
Get-VMSwitch -SwitchType External | Set-VMSwitch -AllowManagementOS $false
ovs-vsctl --no-wait --may-exist add-br br-ex
ovs-vsctl --no-wait add-port br-ex "$INTERFACE_ALIAS"
Get-VMSwitch -SwitchType External | Enable-VMSwitchExtension "Cloudbase Open vSwitch Extension"; sleep 2; Restart-Service ovs-vswitchd
# Clone the MAC Address of $INTERFACE_ALIAS on br-ex
$MAC_ADDRESS=$(Get-NetAdapter "$INTERFACE_ALIAS").MacAddress
$FAKE_MAC_ADDRESS=$MAC_ADDRESS.Substring(0,15)+"99"
Set-NetAdapter -Name "$INTERFACE_ALIAS" -MacAddress $FAKE_MAC_ADDRESS -Confirm:$false
Set-NetAdapter -Name br-ex -MacAddress $MAC_ADDRESS -Confirm:$false
# br-ex will get all the interface details from the DHCP server now
Enable-NetAdapter br-ex

**Comment by aserdean** [comment]
`Get-Vmswitchextension *`
`Get-vmswitch *`

**Comment by mihail-velikov** [comment]
Here is the result:
https://imgur.com/a/FCmb2

**Comment by aserdean** [comment]
Can you disable the extension:
`Disable-vmswitchextension *`
And run:
`Enable-VMSwitchExtension "Cloudbase Open vSwitch Extension"`
and show me the output pls?

**Comment by mihail-velikov** [comment]
I wanted to test something so I restored the machine to a known good state. I will redeploy it and provide detailed logs and screenshots.

**Comment by aserdean** [comment]
Thanks!

**Comment by mihail-velikov** [comment]
Ok, I have redeployed, replayed all steps and here are the logs :)

Here is the status after the packets are installed:
https://imgur.com/a/yAZGs

After running docker network create:
https://imgur.com/a/WAOB4

After running the configuration steps:
https://imgur.com/a/eAmCb

Openvswitch logs after install:
https://imgur.com/a/bxFGA

Network configuration:
https://imgur.com/a/pIAwF

Let me know if you need anything else :)

**Comment by aserdean** [comment]
Sorry I didn't notice you had two vSwitches.
I'll add some more information on the topic:
OVS works only when **enabled on a single vSwitch** and **only one** Hyper-V vSwitch forwarding extension can be enabled at a time.
Mind trying something a bit simpler (Hyper-V route)? Here is a step by step instruction:
```
PS C:\Users\Administrator> Get-VMSwitch

Name SwitchType NetAdapterInterfaceDescription
---- ---------- ------------------------------
nat  Private


PS C:\Users\Administrator> Get-NetAdapter

Name                      InterfaceDescription                    ifIndex Status       MacAddress             LinkSpeed
----                      --------------------                    ------- ------       ----------             ---------
Ethernet0                 Intel(R) PRO/1000 MT Network Connection      11 Up           00-0C-29-58-55-ED         1 Gbps


PS C:\Users\Administrator> ovs-vsctl.exe show
2fb03ee8-51e3-4272-a8a1-30e3ea5454eb
PS C:\Users\Administrator> New-VMSwitch -Name test -NetAdapterName Ethernet0 -AllowManagementOS $false

Name SwitchType NetAdapterInterfaceDescription
---- ---------- ------------------------------
test External   Intel(R) PRO/1000 MT Network Connection


PS C:\Users\Administrator> Get-VMSwitchExtension -Name "open vswitch extension" *


Id                  : 583CC151-73EC-4A6A-8B47-578297AD7623
Name                : Open vSwitch Extension
Vendor              : The Linux Foundation (R)
Version             : 2.9.90
ExtensionType       : Forwarding
ParentExtensionId   :
ParentExtensionName :
SwitchId            : 19f1eb28-41f3-4c71-88a9-e3407ad89808
SwitchName          : test
Enabled             : False
Running             : True
CimSession          : CimSession: .
ComputerName        : WIN-I44SFKIGO35
IsDeleted           : False



PS C:\Users\Administrator> ovs-dpctl.exe show
2018-04-03T11:33:10Z|00001|netlink_socket|ERR|fcntl: The system cannot find the file specified.
2018-04-03T11:33:10Z|00002|dpif|WARN|failed to enumerate system datapaths: Invalid argument
PS C:\Users\Administrator> Enable-VMSwitchExtension "open vswitch extension" test


Id                  : 583CC151-73EC-4A6A-8B47-578297AD7623
Name                : Open vSwitch Extension
Vendor              : The Linux Foundation (R)
Version             : 2.9.90
ExtensionType       : Forwarding
ParentExtensionId   :
ParentExtensionName :
SwitchId            : 19f1eb28-41f3-4c71-88a9-e3407ad89808
SwitchName          : test
Enabled             : True
Running             : True
CimSession          : CimSession: .
ComputerName        : WIN-I44SFKIGO35
IsDeleted           : False



PS C:\Users\Administrator> ovs-dpctl.exe show
system@ovs-system:
        lookups: hit:0 missed:79 lost:0
        flows: 0
PS C:\Users\Administrator> ovs-vsctl.exe add-br br0 -- add-port br0 Ethernet0; Enable-NetAdapter br0
PS C:\Users\Administrator> ovs-dpctl.exe show
system@ovs-system:
        lookups: hit:40 missed:243 lost:0
        flows: 30
        port 1: br0 (internal)
        port 2: Ethernet0
PS C:\Users\Administrator> ovs-vsctl.exe show
2fb03ee8-51e3-4272-a8a1-30e3ea5454eb
    Bridge "br0"
        Port "Ethernet0"
            Interface "Ethernet0"
        Port "br0"
            Interface "br0"
                type: internal
PS C:\Users\Administrator> ipconfig

Windows IP Configuration


Ethernet adapter br0:

   Connection-specific DNS Suffix  . : localdomain
   Link-local IPv6 Address . . . . . : fe80::44ba:65a6:5ec6:e4aa%15
   IPv4 Address. . . . . . . . . . . : 192.168.29.238
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.29.2

Tunnel adapter Local Area Connection* 3:

   Connection-specific DNS Suffix  . :
   IPv6 Address. . . . . . . . . . . : 2001:0:9d38:90d7:2899:176e:aa45:74f2
   Link-local IPv6 Address . . . . . : fe80::2899:176e:aa45:74f2%6
   Default Gateway . . . . . . . . . : ::

Tunnel adapter isatap.localdomain:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . : localdomain
PS C:\Users\Administrator> ping -t google.com

Pinging google.com [172.217.21.238] with 32 bytes of data:
Reply from 172.217.21.238: bytes=32 time=53ms TTL=128
Reply from 172.217.21.238: bytes=32 time=51ms TTL=128

Ping statistics for 172.217.21.238:
    Packets: Sent = 2, Received = 2, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 51ms, Maximum = 53ms, Average = 52ms
Control-C
PS C:\Users\Administrator>
```
Another thing to add is that OVS requires vmports to be named the same on the host (w/o the "vEthernet(" prefix). I.e.:
```
PS C:\Users\Administrator> Get-VMNetworkAdapter -ManagementOS

Name IsManagementOs VMName SwitchName MacAddress   Status IPAddresses
---- -------------- ------ ---------- ----------   ------ -----------
br0  True                  test       00155D1D830F {Ok}


PS C:\Users\Administrator> Get-NetAdapter

Name                      InterfaceDescription                    ifIndex Status       MacAddress             LinkSpeed
----                      --------------------                    ------- ------       ----------             ---------
br0                       Hyper-V Virtual Ethernet Adapter             15 Up           00-15-5D-1D-83-0F         1 Gbps
Ethernet0                 Intel(R) PRO/1000 MT Network Connection      11 Up           00-0C-29-58-55-ED         1 Gbps
```

The steps `##Clone the MAC Address of $INTERFACE_ALIAS on br-ex` are required on cloud providers or any other environment that doesn't allow MAC address spoofing.

I hope that explains things a bit.

Now in your case it looks like the extension hasn't been properly initialized.
Mind running:
`Disable-vmswitchextension *`
`Get-VMSwitch -SwitchType External | Enable-VMSwitchExtension "Cloudbase Open vSwitch Extension"`
And verify via:
`ovs-dpctl show` if the extension was properly initialized?

**Comment by alinbalutoiu** [comment]
@mihail-velikov I think you're hitting the same issue as the other guys. I have also updated the other issue with more details and a fix on it and updated the documentation: https://github.com/openvswitch/ovn-kubernetes/issues/287#issuecomment-378029996

**Comment by mihail-velikov** [comment]
Hello @aserdean,

No luck - no change after running the commands.

This is terribly familiar to me. I think I had the same issue when I initially tried to use 1709 with OVS. 

The problem was that when you install docker it adds a second vmswitch that is used for its default "nat" network. This network can NOT be delete in 1709 because the powershell commandlets were removed. I will try to find the article I read about this.

Thanks a lot!

Kind Regards,
Mihail Velikov

 but can be in 2016 RTM.

and that causes OVS not to work. Unfortunately they remove the functionality

**Comment by alinbalutoiu** [comment]
@mihail-velikov to remove the default ``nat`` network you can run the following:
```
Stop-Service docker
# Get the ID of the nat network, you can use Get-HNSNetwork
Get-HNSNetwork -Id <hns_id> | Remove-HNSNetwork
cmd /c 'echo { "bridge" : "none" } > C:\ProgramData\docker\config\daemon.json'
Start-Service docker
```
On Windows Server RTM you have ``Get-ContainerNetwork`` and ``Remove-ContainerNetwork`` instead of the ones from above.

**Comment by mihail-velikov** [comment]
@alinbalutoiu  - First I tried the command with the pipe. Just to make sure it was executing correctly I got the ID of the switch by hand and then run the second command and replaced the ID manually. There was no difference though.

**Comment by mihail-velikov** [comment]
I have tried the following:
1. Restored the machine to a known good state
2. Stopped the firewall
3. Installed hyper-v
4. installed docker
5. rebooted
6. removed the default docker nat network
7. rebooted
8. Install openvswitch
9. rebooted
10. get-vmswitch was returning empty result
11. manually created a new vmswitch as per @aserdean recommendation
New-VMSwitch -Name test -NetAdapterName Ethernet -AllowManagementOS $false

12. get-vmswitch now shows the "test" switch
13. run:
$INTERFACE_ALIAS="Ethernet"
Stop-Service ovs-vswitchd -force; Get-VMSwitch -SwitchType External | Disable-VMSwitchExtension "Cloudbase Open vSwitch Extension"
Get-VMSwitch -SwitchType External | Set-VMSwitch -AllowManagementOS $false
Get-VMSwitch -SwitchType External | Set-VMSwitch -AllowManagementOS $false
ovs-vsctl --no-wait --may-exist add-br br-ex
ovs-vsctl --no-wait add-port br-ex "$INTERFACE_ALIAS"
Enable-VMSwitchExtension "Cloudbase Open vSwitch Extension" test; sleep 2; Restart-Service ovs-vswitchd

still ovs-dpctl gives the same error message as before

**Comment by aserdean** [comment]
@mihail-velikov I'm really sorry it is not working for you.
Can you post the output of the following:
```wmic os get buildnumber
wmic qfe list```
I will replicate your environment and try to figure out what is going on.

**Comment by mihail-velikov** [comment]
Hello @aserdean 

Please excuse me for my late reply. Here is the output of the commands:
![screen shot 2018-04-10 at 9 25 18](https://user-images.githubusercontent.com/13517115/38539734-fc770596-3ca1-11e8-98c1-0fa960058c41.png)

In the mean time I will download the latest insider preview version of Windows and run it in a virtulbox on my local machine to try and execute the same steps.

Let me know if you need anything else.

Thank you.

Kind Regards,
Mihail velikov

**Comment by mihail-velikov** [comment]
Hello again,

So I did a bit of testing:
1. Downloaded the latest windows insider build - 17623 (windows 2019)
2. Installed it locally
3. Removed the firewall
4. Installed hyper-v
5. Restarted the machine
6. Downloaded the  https://cloudbase.it/downloads/openvswitch-hyperv-installer-beta.msi and installed the package
7. restared the machine
8. Run ovs-vsctl.exe -V 
2.7.4
9. Tried to manually add a new virtual switch using this commands:
 New-VMSwitch -Name test -NetAdapterName Ethernet -AllowManagementOS $false
10. The switch was added
11. Enabled the "Cloudbase Open vSwitch Extension" for the switch test
12. It reported as enabled and running
13. run ovs-dpctl show but it still reported that:
netlink_socket|ERR|fcntl: The system cannot find the file specified.
dpif|WARN|failed to enumerate system datapaths: Invalid argument

One thing I noticed is that in the steps provided by @aserdean , the name of his extension and version is different - the name is "Open vSwitch Extension" and the version was 2.9.90

In my case the name is "Cloudbase Open vSwitch Extension" and the version is 2.7.4

Any thoughts?

Kind Regards,
Mihail Velikov

**Comment by mihail-velikov** [comment]
Hello again,

I have tried the same steps as above with the only difference that instead of using this package:
https://cloudbase.it/downloads/openvswitch-hyperv-installer-beta.msi
I used this one:
https://cloudbase.it/downloads/openvswitch-hyperv-2.7.0-certified.msi

This works successfully and the br-ex interface is created and working.

I guess there is some kind of a problem with the beta installer for openvswitch.

Thank you.

Kind Regards,
Mihail Velikov

**Comment by aserdean** [comment]
Thanks a lot for the work @mihail-velikov .
I'll deploy the latest insider preview and will look into it this week.

We added some ansible playbooks to ease up deployments: https://github.com/openvswitch/ovn-kubernetes/pull/302 .
Do you mind trying them out?

**Comment by mihail-velikov** [comment]
Hey @aserdean,

It seems I forgot to post a reply.  I have written our own ansible code for provisioning our cluster due to some limitation that we have in our datacenters.

Btw I have already moved our cluster to windows 2019 preview and testing the stability, so .. wish me luck :)

Kind Regards,
Mihail Velikov

**Comment by shettyg** [comment]
The original issue is fixed with the released OVS 2.9.1
